docs/guide/cli/index.md

```diff
@@ -0,0 +1,253 @@
+```
+
+**Arguments:**
+- `path` (optional) - Path to a directory or tarball file to analyze. Defaults to current directory.
+
+**Options:**
+- `--pack <type>` - Package manager to use for packing. Options: `auto`, `npm`, `yarn`, `pnpm`, `bun`, `none`. Default: `auto`
+- `--log-level <level>` - Set the log level. Options: `debug`, `info`, `warn`, `error`. Default: `info`
+- `--manifest <path>` - Path(s) to custom manifest file(s) for module replacements analysis
+
+**Examples:**
+```sh
+# Analyze current project
+e18e-cli analyze
+
+# Analyze with debug output
+e18e-cli analyze --log-level debug
+
+# Analyze with specific package manager
+e18e-cli analyze --pack pnpm
+```
+
+### `migrate`
+
+Migrates your project from packages to more performant alternatives.
+
+```bash
+e18e-cli migrate [packages...] [options]
+```
+
+**Arguments:**
+- `packages...` - Names of packages to migrate (e.g., `chalk`, `lodash`)
+
+**Options:**
+- `--all` - Run all available migrations
+- `--dry-run` - Don't apply any fixes, only show what would change
+- `--include <pattern>` - Files to migrate. Default: `**/*.{ts,js}`
+- `--interactive` - Run in interactive mode
+- `--manifest <path>` - Use custom migration manifest file(s)
+
+**Examples:**
+```bash
+# Migrate specific packages
+e18e-cli migrate chalk lodash
+
+# Preview changes without applying
+e18e-cli migrate chalk --dry-run
+
+# Migrate all available packages
+e18e-cli migrate --all
+
+# Interactive mode to select packages
+e18e-cli migrate --interactive
+
+# Use custom migration manifest
+e18e-cli migrate --manifest ./custom-migrations.json
+```
+
+## Migration System
+
+The CLI uses an [opinionated codemod](https://github.com/es-tooling/module-replacements-codemods) approach for migrations, based on e18e's curated list of performance-focused package replacements. This ensures consistent, well-tested migrations across the ecosystem.
+
+### Default Migrations
+The CLI comes with pre-configured migrations for common performance optimizations:
+- `chalk` → `picocolors` (smaller, faster)
+- `lodash` → `lodash-es` (ESM version)
+- `moment` → `date-fns` (tree-shakeable)
+- `axios` → `undici` (native fetch-like API)
+
+> [!IMPORTANT]
+> The replacement list is still being developed and expanded. The current list represents the initial set of well-tested migrations. More replacements will be added as they are validated and tested across the ecosystem.
+
+### Custom Manifests
+You can bring your own migration rules by providing custom manifest files:
+
+```json
+{
+  "replacements": [
+    {
+      "from": "old-package",
+      "to": "new-package",
+      "reason": "Better performance and smaller bundle"
+    }
+  ]
+}
+```
+
+Use custom manifests with the `--manifest` option:
+```bash
+e18e-cli migrate --manifest ./my-migrations.json
+```
+
+## What the CLI Analyzes
+
+The `analyze` command runs several checks on your project:
+
+1. **ATTW (Are The Types Wrong)** - Checks for TypeScript declaration file issues
```

Comment:
instead of listing the underlying tools we use, i think we should probably list the checks and link to the tool

something like this:

> 1. Check for TypeScript declaration issues (using [attw](/))
> 2. Validate package.json and publishing configuration (using [publint](/))
> 3. Identify opportunities to replace packages with more performant alternatives (using [module-replacements](/))
> 4. Analyze your dependency tree for optimization opportunities